src/Inkluzitron/Services/TypeReaders/TimeSpanTypeReader.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Discord.Commands;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Inkluzitron.Services.TypeReaders
+{
+    public class TimeSpanTypeReader : TypeReader
+    {
+        private const RegexOptions regexOptions = RegexOptions.IgnoreCase | RegexOptions.Multiline | RegexOptions.CultureInvariant | RegexOptions.IgnorePatternWhitespace;
+
+        private Dictionary<Regex, Func<Match, TimeSpan>> MatchingFunctions { get; } = new()
+        {
+            { new Regex(@"(\d+)\s*(dn[ůíy]?|d)", regexOptions), m => TimeSpan.FromDays(int.Parse(m.Groups[1].Value)) },
+            { new Regex(@"(\d+)\s*(hodin[auy]?|hod|h)", regexOptions), m => TimeSpan.FromHours(int.Parse(m.Groups[1].Value)) },
+            { new Regex(@"(\d+)\s*(minut[auy]?|min|m)", regexOptions), m => TimeSpan.FromMinutes(int.Parse(m.Groups[1].Value)) },
+            { new Regex(@"(\d+)\s*(sekund[auy]?|sec|s)", regexOptions), m => TimeSpan.FromSeconds(int.Parse(m.Groups[1].Value)) },
+
+            { new Regex(@"(^|\s+)(1\s+den|den|dny)($|\s+)", regexOptions), _ => TimeSpan.FromDays(1) },
```

Comment:
Možná se pletu, ale neudělá to match i na `1 dny`? :hmmnote:

Reply:
Trochu jsem to jeste poupravil, ale mam tuseni, ze ke zpracovani prirozeneho jazyka se muzeme priblizovat donekonecna